### ISSUE TITLE
Songs not completing

### ISSUE THREAD

**Issue by jlwright325** [joek13/py-music-bot#9]
Songs don't seem to be completing fully. They make it about halfway and then stop playing or continue to next song.

**Comment by BatMuncher69** [comment]
@jlwright325 i have the same issue

**Comment by joek13** [comment]
Hey @jlwright325 and @BatMuncher69, thanks for taking the time to come leave some feedback on my project.

I haven't looked into this issue yet, but my suspicion is that it might have something to do with streams closing unexpectedly. This may involve looking more deeply into how the bot uses `ffmpeg`. When I have a little bit more time, I'll dig in and figure out if I can get a fix pushed. Thanks for your patience. I'll post updates in this issue.

**Comment by b4skyx** [comment]
I'm having the same issue. The error I get is:
```
[tls @ 0x55c7b0f75b40] 
Error in the pull function. 
[matroska,webm @ 0x55c7b0f71740] 
Read error [tls @ 0x55c7b0f75b40] 
The specified session has been invalidated for some reason.
```

**Comment by avy9999** [comment]
Did U Guys Found any solution regarding that?

**Comment by RedKrieg** [comment]
Hi all, I was having the same issue and I think I fixed it in my latest pull request.  Logs from before vs after:

Before
----

```
Apr 13 15:26:13 valheim-oss yukevox[2573864]: [2021-04-13 15:26:13,118] INFO [root.play:259] Now playing 'While The World Sleeps'
Apr 13 15:34:21 valheim-oss yukevox[2775646]: [tls @ 0x55c92b9dcb00] Error in the pull function.
Apr 13 15:34:21 valheim-oss yukevox[2775646]: [matroska,webm @ 0x55c92b89a700] Read error
Apr 13 15:34:21 valheim-oss yukevox[2775646]: [tls @ 0x55c92b9dcb00] The specified session has been invalidated for some reason.
Apr 13 15:34:21 valheim-oss yukevox[2775646]:     Last message repeated 1 times
Apr 13 15:34:21 valheim-oss yukevox[2775646]: https://r3---sn-xxxxxxxxx.googlevideo.com/videoplayback?xxxxxxxxxxxxxxxxxxxxxxxxx: Input/output error
Apr 13 15:34:21 valheim-oss yukevox[2573864]: [2021-04-13 15:34:21,854] INFO [discord.player.cleanup:158] Preparing to terminate ffmpeg process 2775646.
Apr 13 15:34:21 valheim-oss yukevox[2573864]: [2021-04-13 15:34:21,855] INFO [discord.player.cleanup:170] ffmpeg process 2775646 successfully terminated with return code of 0.
```

After
----

```
Apr 13 16:11:45 valheim-oss yukevox[2779026]: [2021-04-13 16:11:45,151] INFO [root.play:259] Now playing 'While The World Sleeps'
Apr 13 16:19:27 valheim-oss yukevox[2779437]: [tls @ 0x55d9c1befc80] Error in the pull function.
Apr 13 16:19:27 valheim-oss yukevox[2779437]: [https @ 0x55d9c1bed5c0] Will reconnect at 3080176 in 0 second(s), error=Input/output error.
Apr 13 16:20:04 valheim-oss yukevox[2779437]: [tls @ 0x55d9c1ecc080] Error in the pull function.
Apr 13 16:20:04 valheim-oss yukevox[2779437]: [https @ 0x55d9c1bed5c0] Will reconnect at 3375088 in 0 second(s), error=Input/output error.
Apr 13 16:20:20 valheim-oss yukevox[2779026]: [2021-04-13 16:20:20,354] INFO [discord.player.cleanup:158] Preparing to terminate ffmpeg process 2779437.
Apr 13 16:20:20 valheim-oss yukevox[2779026]: [2021-04-13 16:20:20,355] INFO [discord.player.cleanup:170] ffmpeg process 2779437 successfully terminated with return code of 0.
```

**Comment by Blueman678** [comment]
> Hi all, I was having the same issue and I think I fixed it in my latest pull request. Logs from before vs after:
> 
> ## Before
> ```
> Apr 13 15:26:13 valheim-oss yukevox[2573864]: [2021-04-13 15:26:13,118] INFO [root.play:259] Now playing 'While The World Sleeps'
> Apr 13 15:34:21 valheim-oss yukevox[2775646]: [tls @ 0x55c92b9dcb00] Error in the pull function.
> Apr 13 15:34:21 valheim-oss yukevox[2775646]: [matroska,webm @ 0x55c92b89a700] Read error
> Apr 13 15:34:21 valheim-oss yukevox[2775646]: [tls @ 0x55c92b9dcb00] The specified session has been invalidated for some reason.
> Apr 13 15:34:21 valheim-oss yukevox[2775646]:     Last message repeated 1 times
> Apr 13 15:34:21 valheim-oss yukevox[2775646]: https://r3---sn-xxxxxxxxx.googlevideo.com/videoplayback?xxxxxxxxxxxxxxxxxxxxxxxxx: Input/output error
> Apr 13 15:34:21 valheim-oss yukevox[2573864]: [2021-04-13 15:34:21,854] INFO [discord.player.cleanup:158] Preparing to terminate ffmpeg process 2775646.
> Apr 13 15:34:21 valheim-oss yukevox[2573864]: [2021-04-13 15:34:21,855] INFO [discord.player.cleanup:170] ffmpeg process 2775646 successfully terminated with return code of 0.
> ```
> 
> ## After
> ```
> Apr 13 16:11:45 valheim-oss yukevox[2779026]: [2021-04-13 16:11:45,151] INFO [root.play:259] Now playing 'While The World Sleeps'
> Apr 13 16:19:27 valheim-oss yukevox[2779437]: [tls @ 0x55d9c1befc80] Error in the pull function.
> Apr 13 16:19:27 valheim-oss yukevox[2779437]: [https @ 0x55d9c1bed5c0] Will reconnect at 3080176 in 0 second(s), error=Input/output error.
> Apr 13 16:20:04 valheim-oss yukevox[2779437]: [tls @ 0x55d9c1ecc080] Error in the pull function.
> Apr 13 16:20:04 valheim-oss yukevox[2779437]: [https @ 0x55d9c1bed5c0] Will reconnect at 3375088 in 0 second(s), error=Input/output error.
> Apr 13 16:20:20 valheim-oss yukevox[2779026]: [2021-04-13 16:20:20,354] INFO [discord.player.cleanup:158] Preparing to terminate ffmpeg process 2779437.
> Apr 13 16:20:20 valheim-oss yukevox[2779026]: [2021-04-13 16:20:20,355] INFO [discord.player.cleanup:170] ffmpeg process 2779437 successfully terminated with return code of 0.
> ```

yes, it works! thanks a lot...

**Comment by k-rajat42** [comment]
> Hi all, I was having the same issue and I think I fixed it in my latest pull request. Logs from before vs after:
> 
> ## Before
> ```
> Apr 13 15:26:13 valheim-oss yukevox[2573864]: [2021-04-13 15:26:13,118] INFO [root.play:259] Now playing 'While The World Sleeps'
> Apr 13 15:34:21 valheim-oss yukevox[2775646]: [tls @ 0x55c92b9dcb00] Error in the pull function.
> Apr 13 15:34:21 valheim-oss yukevox[2775646]: [matroska,webm @ 0x55c92b89a700] Read error
> Apr 13 15:34:21 valheim-oss yukevox[2775646]: [tls @ 0x55c92b9dcb00] The specified session has been invalidated for some reason.
> Apr 13 15:34:21 valheim-oss yukevox[2775646]:     Last message repeated 1 times
> Apr 13 15:34:21 valheim-oss yukevox[2775646]: https://r3---sn-xxxxxxxxx.googlevideo.com/videoplayback?xxxxxxxxxxxxxxxxxxxxxxxxx: Input/output error
> Apr 13 15:34:21 valheim-oss yukevox[2573864]: [2021-04-13 15:34:21,854] INFO [discord.player.cleanup:158] Preparing to terminate ffmpeg process 2775646.
> Apr 13 15:34:21 valheim-oss yukevox[2573864]: [2021-04-13 15:34:21,855] INFO [discord.player.cleanup:170] ffmpeg process 2775646 successfully terminated with return code of 0.
> ```
> 
> ## After
> ```
> Apr 13 16:11:45 valheim-oss yukevox[2779026]: [2021-04-13 16:11:45,151] INFO [root.play:259] Now playing 'While The World Sleeps'
> Apr 13 16:19:27 valheim-oss yukevox[2779437]: [tls @ 0x55d9c1befc80] Error in the pull function.
> Apr 13 16:19:27 valheim-oss yukevox[2779437]: [https @ 0x55d9c1bed5c0] Will reconnect at 3080176 in 0 second(s), error=Input/output error.
> Apr 13 16:20:04 valheim-oss yukevox[2779437]: [tls @ 0x55d9c1ecc080] Error in the pull function.
> Apr 13 16:20:04 valheim-oss yukevox[2779437]: [https @ 0x55d9c1bed5c0] Will reconnect at 3375088 in 0 second(s), error=Input/output error.
> Apr 13 16:20:20 valheim-oss yukevox[2779026]: [2021-04-13 16:20:20,354] INFO [discord.player.cleanup:158] Preparing to terminate ffmpeg process 2779437.
> Apr 13 16:20:20 valheim-oss yukevox[2779026]: [2021-04-13 16:20:20,355] INFO [discord.player.cleanup:170] ffmpeg process 2779437 successfully terminated with return code of 0.
> ```

thanks for providing a solution but can u tell where to make this exact edit in which file and line ?

**Comment by RedKrieg** [comment]
@k-rajat42 You can find the diff in my PR linked above.  Here's the direct link to the diff that shows exactly what I changed and on what line: https://github.com/joek13/py-music-bot/pull/12/commits/aa9b6c12aba1a5660fad224e1d1e59ea13b55836

**Comment by joek13** [comment]
This should have been fixed by #13. Please comment on this issue if the problem persists.
